uefi/src/proto/device_path/mod.rs

```diff
@@ -488,6 +517,15 @@ impl PartialEq for DevicePath {
     }
 }
 
+impl<'a> TryFrom<&[u8]> for &'a DevicePath {
+    type Error = ByteConversionError;
+
+    fn try_from(bytes: &[u8]) -> Result<Self, Self::Error> {
+        DevicePath::size_in_bytes_from_slice(bytes)?;
+        unsafe { Ok(DevicePath::from_ffi_ptr(bytes.as_ptr().cast())) }
```

Comment:
Calling `DevicePath::from_ffi_ptr` will (indirectly) iterate through the whole device path again to get the length. Instead, construct the reference directly with something like `ptr_meta::from_raw_parts(bytes.as_ptr().cast(), len)`